doc/source/user_guide/io.rst

```diff
@@ -1254,6 +1254,8 @@ The bad line will be a list of strings that was split by the ``sep``:
 
     .. versionadded:: 1.4.0
 
+Note that the callable function will handle only a line with too many fields.
+Bad lines caused by other errors will be silently skipped.
```

Comment:
Do you mind adding a paragraph with code examples with that `on_bad_lines="warn"|"error"|callable` does with a different definition of bad line (e.g. like the example in the original issue)?

Reply:
@kostyafarber do you have time to address this? I think the comment you are adding is quite useful, but it'll be even better if you can add the example at least with `on_bad_lines='error'` as suggested, and also provide more information on what are the other errors that will be skipped, and an example of them.

Thanks!

Reply:
Happy to work on this. Have been busy over the holidays. Will pick this up and get it down in the next week or so.

Thanks for chasing up.

Reply:
Thanks @kostyafarber, this looks good to me.

I personally find this behavior an obvious bug, and move forward and report as bad line anything that can't be parsed. Silently ignoring errors go against the zen of Python, general programming practices, and makes user life much more difficult. But if there is no agreement to change the behavior at this point (I think there should be), happy with the doc update.

Reply:
I absolutely agree, the issue has been left open to discuss further improvements (expand definition of a bad line). 

I guess in the interim it would be good to let users know of this behaviour.